--- gettext.py.old	2006-09-07 16:12:29.000000000 +0200
+++ gettext.py	2006-09-07 16:12:41.000000000 +0200
@@ -126,8 +126,11 @@
 
 
 def _expand_lang(locale):
-    from locale import normalize
-    locale = normalize(locale)
+    try:
+        from locale import normalize
+        locale = normalize(locale)
+    except ImportError:
+        pass
     COMPONENT_CODESET   = 1 << 0
     COMPONENT_TERRITORY = 1 << 1
     COMPONENT_MODIFIER  = 1 << 2
