Index: run_testcase.py
===================================================================
--- run_testcase.py	(révision 916)
+++ run_testcase.py	(copie de travail)
@@ -249,7 +249,7 @@
     if len(sys.argv) > 1:
         directory = sys.argv[1]
     else:
-        directory = os.path.join(os.path.dirname(__file__), "testcase")
+        directory = "testcase"
     charset = getTerminalCharset()
     directory = unicode(directory, charset)
     print "Start testcase."
Index: hachoir/field/integer.py
===================================================================
--- hachoir/field/integer.py	(révision 904)
+++ hachoir/field/integer.py	(copie de travail)
@@ -50,7 +50,6 @@
                 self.getOriginalDisplay = self._getDisplay
                 self._getDisplay = lambda: text_handler(self)
     cls = Integer
-    cls.__name__ = name
     return cls

 UInt8 = integerFactory("UInt8", False, 8)
Index: hachoir/field/string_field.py
===================================================================
--- hachoir/field/string_field.py	(révision 919)
+++ hachoir/field/string_field.py	(copie de travail)
@@ -247,7 +247,6 @@
             GenericString.__init__(self, parent, name, format, description,
             strip=strip, charset=charset)
     cls = NewString
-    cls.__name__ = name
     return cls

 # String which ends with nul byte ("\0")
